src/main/java/com/inventoryservice/controller/CategoryController.java

```diff
@@ -0,0 +1,187 @@
+import com.inventoryservice.exception.CategoryNotFoundException;
+import com.inventoryservice.exception.SubCategoryNotFoundException;
+import com.inventoryservice.model.*;
+import com.inventoryservice.service.CategoryService;
+import com.inventoryservice.service.SubCategoryService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import java.util.List;
+import java.util.stream.Collectors;
+
+
+@RestController
+@RequestMapping("/category")
+public class CategoryController {
+
+    @Autowired
+    private CategoryService categoryService;
+
+    @Autowired
+    private SubCategoryService subcategoryService;
+
+    /*  To Add new Category*/
+    @PostMapping()
+    @PermitInventoryManager
+    public ResponseEntity<?> createCategory(@RequestBody CategoryRequest categoryRequest) {
+        Category category = categoryRequest.toEntity(categoryRequest);
+        categoryService.createCategory(category);
+        return new ResponseEntity<>(category, HttpStatus.FOUND);
+    }
+
+    /*  To find Category
+     * @param  categoryId an Id giving the categoryId of specific Id
+     * @return      Category of specific Id.
+     * @throws CategoryNotFoundException  If an Category of specific Id is not found
+     * */
+    @GetMapping("/{categoryId}")
+    public ResponseEntity<?> findCategoryById(@PathVariable int categoryId) throws CategoryNotFoundException {
+        Category category = categoryService.findCategoryById(categoryId);
+        if (category == null) {
+            throw new CategoryNotFoundException("Category not found");
+        }
+        return new ResponseEntity<>(category, HttpStatus.FOUND);
+    }
+
+    /*  To Delete Category
+     * @param  categoryId an Id giving the Category of specific Id
+     * @return it deleted Category of specific Id.
+     * @throws CategoryNotFoundException  If an Category of specific Id is not found
+     */
+    @DeleteMapping("/{categoryId}")
+    @PermitInventoryManager
+    public ResponseEntity<?> deleteCategory(@PathVariable int categoryId) throws CategoryNotFoundException {
+        Category category = categoryService.findCategoryById(categoryId);
+        if (category == null) {
+            throw new CategoryNotFoundException("Category not found");
+        }
+        if (!category.isArchived()) {
+            categoryService.deleteCategory(categoryId);
+            return new ResponseEntity<String>("Category Deleted successfully", HttpStatus.FOUND);
+        }
+        return new ResponseEntity<String>("Category is archived and cannot be deleted", HttpStatus.BAD_REQUEST);
+    }
+
+    /*
+     * To get list of all Category
+     * @return list of all Category .
+     */
+    @GetMapping()
+    public ResponseEntity<List<Category>> getAllCategories() {
+        List<Category> categories = categoryService.getAllCategories();
```

Comment:
We shouldn't use model classes for sending response to client.

Reply:
> We shouldn't use model classes for sending response to client.

Completed and used Response class for sending response to client